.github/workflows/docker.yml

```diff
@@ -0,0 +1,71 @@
+
+name: Docker CI/CD
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - '*'
+  pull_request:
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Log in to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: ghcr.io/${{ github.repository }}
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v4
+        with:
+          context: .
+          push: true
+          tags: ${{ github.ref_name }}
+
+  deploy:
+    runs-on: ubuntu-latest
+    needs: build-and-push
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Log in to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Create .env file
+        run: |
+          echo "IMAGE_NAME=ghcr.io/${{ github.repository }}" > .env
+          echo "IMAGE_TAG=${{ github.ref_name }}" >> .env
+          echo "CONTAINER_NAME=deriv-api-bff-${{ github.ref_name }}" >> .env
+          echo "HOST_PORT=8080" >> .env
+          echo "GLOBAL_DEBUG_ON=true" >> .env
+
+      - name: Start Docker Compose
+        run: ./scripts/start_container.sh
+
+      - name: Stop Docker Compose
+        run: ./scripts/stop_container.sh
```

Comment:
@shan-96 I think it'll never fail, I think these 2 steps doesn't provide a value

Reply:
yes right now just focusing on starting and stopping the server